core/src/main/java/org/bitcoinj/wallet/DeterministicKeyChain.java

```diff
@@ -309,7 +309,7 @@ protected DeterministicKeyChain(DeterministicSeed seed) {
      * this method to watch an arbitrary fragment of some other tree, this limitation may be removed in future.
      */
     public DeterministicKeyChain(DeterministicKey watchingKey, long creationTimeSeconds) {
-        checkArgument(watchingKey.isPubKeyOnly(), "Private subtrees not currently supported");
```

Comment:
Maybe we need to forbid getPubOnly on non-root HD keys. To get a pubOnly HD key you'd need to pubOnly the root and re-derive the hierarchy as needed.